### DISCUSSION TITLE
index.pageContext.json added to urlOriginal on navigate client side

### DISCUSSION THREAD

**Discussion by royalswe** [vikejs/vike#931] [Discussion] [answered]
When i do client routing on my Vue app it ads `index.pageContext.json` to the route path.

Bellow is logs from node server when load wood page and the navigate to think-wood as client routing.
```
14:20:18 [vite-plugin-ssr] HTTP Request 200 /sv/se/wood/
14:20:35 [vite-plugin-ssr] HTTP Request 200 /sv/se/wood/think-wood/index.pageContext.json
```
I can se the `index.pageContext.json` at the end when I log out the pageContext.urlOriginal string.
The url in the browser is not affected.

Why is that?

**Comment by brillout** [comment, marked as answer]
That's how VPS fetches the data returned by `onBeforeRender()`. If you don't want that: https://vite-plugin-ssr.com/migration/v1-design#onbeforerender-in-page-js.
